src/services/safe_border.py

```diff
@@ -179,22 +176,20 @@ def _find_earliest_slashed_epoch_rounded_to_frame(self, validators: list[Validat
         """
         Returns the earliest slashed epoch for the given validators rounded to the frame
         """
-        withdrawable_epoch = min(get_validators_withdrawable_epochs(validators))
         last_finalized_request_id_epoch = self.get_epoch_by_slot(self._get_last_finalized_withdrawal_request_slot())
-
-        earliest_activation_slot = self._get_validators_earliest_activation_epoch(validators)
-        max_possible_earliest_slashed_epoch = EpochNumber(withdrawable_epoch - EPOCHS_PER_SLASHINGS_VECTOR)
-
         # Since we are looking for the safe border epoch, we can start from the last finalized withdrawal request epoch
         # or the earliest activation epoch among the given validators for optimization
-        start_epoch = max(last_finalized_request_id_epoch, earliest_activation_slot)
+        earliest_activation_epoch = min((v.validator.activation_epoch for v in validators), default=0)
+        start_epoch = max(last_finalized_request_id_epoch, earliest_activation_epoch)
 
         # We can stop searching for the slashed epoch when we reach the reference epoch
         # or the max possible earliest slashed epoch for the given validators
+        withdrawable_epoch = min(v.validator.withdrawable_epoch for v in validators)
```

Comment:
Let's check that validators is not empty list, or add default here

Reply:
I'd prefer to raise the exception at the beginning of the function in case of empty list